pkg/controller/machine_safety.go

```diff
@@ -543,30 +543,33 @@ func (c *controller) unfreezeMachineSetsAndDeployments(machineSet *v1alpha1.Mach
 		glog.Warning("Updated failed, retrying - ", err)
 	}
 
-	machineDeployment := c.getMachineDeploymentsForMachineSet(machineSet)[0]
-	if machineDeployment != nil {
-		for {
-			// Get the latest version of the machineDeployment so that we can avoid conflicts
-			machineDeployment, err := c.controlMachineClient.MachineDeployments(machineDeployment.Namespace).Get(machineDeployment.Name, metav1.GetOptions{})
-			if err != nil {
-				// Some error occued while fetching object from API server
-				glog.Error(err)
-				break
-			}
-			clone := machineDeployment.DeepCopy()
-			if clone.Labels == nil {
-				clone.Labels = make(map[string]string)
-			}
-			newStatus := clone.Status
-			RemoveMachineDeploymentCondition(&newStatus, v1alpha1.MachineDeploymentFrozen)
-			clone.Status = newStatus
-			delete(clone.Labels, "freeze")
-			_, err = c.controlMachineClient.MachineDeployments(clone.Namespace).Update(clone)
-			if err == nil {
-				break
+	machineDeployments := c.getMachineDeploymentsForMachineSet(machineSet)
```

Comment:
Wondering why do we need to replace the existing way of fetching the MachineDeployment?
Is there any rationale or is it just being considered as a good practice?

Reply:
The logic is simply enhanced with some null checks, it is a good practice. Either we enforce that MachineDeployment cannot be null for MachineSet or do a null check while fetching the same

Reply:
I don't see the existing way is changed. Only additional length and nil checks. The diff tool seems to be not up to the task to show actual change here :-)